From b8e892d2359321cdc7793af0c25948c543a95ba3 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Wed, 13 Jul 2022 22:47:27 +0200
Subject: [PATCH] strace: update to version 5.16

Fixes:
D_FORTIFY_SOURCE=2 -Wl,-z,now -Wl,-z,relro  -MT libstrace_a-net.o -MD -MP -MF .deps/libstrace_a-net.Tpo -c -o libstrace_a-net.o `test -f 'net.c' || echo './'`net.c
net.c:35:11: fatal error: linux/ipx.h: No such file or directory
 # include <linux/ipx.h>
           ^~~~~~~~~~~~~
compilation terminated.

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
---
 package/devel/strace/Makefile | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/package/devel/strace/Makefile b/package/devel/strace/Makefile
index a95d429311..8f66ffbe47 100644
--- a/package/devel/strace/Makefile
+++ b/package/devel/strace/Makefile
@@ -9,12 +9,12 @@ include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=strace
-PKG_VERSION:=5.10
+PKG_VERSION:=5.16
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=https://strace.io/files/$(PKG_VERSION)
-PKG_HASH:=fe3982ea4cd9aeb3b4ba35f6279f0b577a37175d3282be24b9a5537b56b8f01c
+PKG_HASH:=dc7db230ff3e57c249830ba94acab2b862da1fcaac55417e9b85041a833ca285
 
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
 PKG_LICENSE:=LGPL-2.1-or-later
@@ -70,7 +70,8 @@ endef
 CONFIGURE_ARGS += \
 	--with-libdw=$(if $(CONFIG_STRACE_LIBDW),yes,no) \
 	--with-libunwind=$(if $(CONFIG_STRACE_LIBUNWIND),yes,no) \
-	--enable-mpers=no
+	--enable-mpers=no \
+	--without-libselinux
 
 MAKE_FLAGS := \
 	CCOPT="$(TARGET_CFLAGS)"
-- 
2.34.1

